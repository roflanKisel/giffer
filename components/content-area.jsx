import React from 'react';
import PropTypes from 'prop-types';
import { lifecycle, onlyUpdateForKeys } from 'recompose';
import styled from 'styled-components';
import HoverPreview from './hover-preview';
import withLoadingStatus from './loading-status';

export const ContentArea = ({ contentList }) => (
  <ContentAreaWrapper>
    {contentList &&
      contentList.map(content => (
        <HoverPreview
          key={content.id}
          id={content.id}
          src={content.url}
          preview={content.previewUrl}
          width={content.width}
          height={content.height}
          title={content.title}
        />
      ))}
  </ContentAreaWrapper>
);

const ContentAreaWrapper = styled.div`
  display: flex;
  justify-content: space-around;
  flex-wrap: wrap;
  width: 80%;
  z-index: 1;
`;

ContentArea.propTypes = {
  contentList: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      url: PropTypes.string,
      previewUrl: PropTypes.string,
      width: PropTypes.string,
      height: PropTypes.string,
      title: PropTypes.string,
    })
  ).isRequired,
};

const ExtendedContentArea = onlyUpdateForKeys(['contentList'])(ContentArea);

export const ContentAreaWithLoadingStatus = withLoadingStatus(
  ExtendedContentArea
);

export const ContentAreaWithMounting = lifecycle({
  componentDidMount() {
    const { handleTrendingSearch } = this.props;

    handleTrendingSearch();
  },
})(ContentAreaWithLoadingStatus);

export default ContentAreaWithMounting;
