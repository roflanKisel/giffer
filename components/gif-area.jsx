import React from 'react';
import styled from 'styled-components';
import { lifecycle } from 'recompose';
import PropTypes from 'prop-types';
import withLoadingStatus from './loading-status';

const GifArea = ({ gif }) => (
  <div>
    {gif && (
      <StyledImageWrapper>
        <StyledImage src={gif.url} height={gif.height} width={gif.width} />
      </StyledImageWrapper>
    )}
  </div>
);

GifArea.defaultProps = {
  gif: null,
};

GifArea.propTypes = {
  gif: PropTypes.shape({
    url: PropTypes.string,
    height: PropTypes.string,
    width: PropTypes.string,
  }),
};

const StyledImageWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 20px;
  padding: 20px;
`;

const StyledImage = styled.img`
  border-radius: 10px;
`;

const GifAreaWithMounting = lifecycle({
  componentDidMount() {
    const { handleSearchById, gifId } = this.props;

    handleSearchById({ gifId });
  },
})(withLoadingStatus(GifArea));

export default GifAreaWithMounting;
