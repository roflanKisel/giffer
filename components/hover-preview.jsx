import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Link } from '../routes';
import Preview from './preview';

class HoverPreview extends React.PureComponent {
  static propTypes = {
    id: PropTypes.string.isRequired,
    src: PropTypes.string.isRequired,
    preview: PropTypes.string.isRequired,
    height: PropTypes.string.isRequired,
    width: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
  };

  state = {
    isSelected: false,
  };

  onSelectedSet = () => {
    this.setState({
      isSelected: true,
    });
  };

  onSelectedUnset = () => {
    this.setState({
      isSelected: false,
    });
  };

  render() {
    const { src, preview, height, width, id, title } = this.props;
    const { isSelected } = this.state;

    return (
      <Link route="gifpage" params={{ id }}>
        <ImageWrapper
          onMouseEnter={this.onSelectedSet}
          onMouseLeave={this.onSelectedUnset}
        >
          <Preview
            src={isSelected ? src : preview}
            height={height}
            width={width}
            alt={title}
            selected={isSelected}
          />
        </ImageWrapper>
      </Link>
    );
  }
}

const ImageWrapper = styled.a`
  padding: 0px;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export default HoverPreview;
