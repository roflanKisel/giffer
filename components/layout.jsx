import React from 'react';
import Navbar from './navbar';

const withLayout = WrappedComponent =>
  class Layout extends React.Component {
    static getInitialProps(ctx) {
      return WrappedComponent.getInitialProps
        ? WrappedComponent.getInitialProps(ctx)
        : {};
    }

    render() {
      return (
        <div>
          <Navbar />
          <WrappedComponent {...this.props} />
        </div>
      );
    }
  };

export default withLayout;
