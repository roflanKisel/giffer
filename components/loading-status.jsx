import React from 'react';
import styled from 'styled-components';

const withLoadingStatus = Component => ({ isLoading, isFailure, ...props }) => (
  <StyledLoadingStatus>
    {isLoading && !isFailure && <div>Loading...</div>}
    {isFailure && !isLoading && <div>Failured</div>}
    {!isLoading && !isFailure && <Component {...props} />}
  </StyledLoadingStatus>
);

const StyledLoadingStatus = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  max-width: inherit;
`;

export default withLoadingStatus;
