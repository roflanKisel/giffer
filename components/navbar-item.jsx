import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { onlyUpdateForKeys } from 'recompose';
import { Link } from '../routes';

export const NavbarItem = ({ name, link }) => (
  <Link route={link}>
    <StyledNavbarItem>{name}</StyledNavbarItem>
  </Link>
);

const StyledNavbarItem = styled.span`
  padding: 10px 15px 10px 15px;
  cursor: pointer;
  color: white;
  font-size: 19px;
  transition: 0.3s;

  &:hover {
    background-color: #315591;
  }
`;

NavbarItem.defaultProps = {
  name: '',
};

NavbarItem.propTypes = {
  name: PropTypes.string,
  link: PropTypes.string.isRequired,
};

export default onlyUpdateForKeys(['name', 'link'])(NavbarItem);
