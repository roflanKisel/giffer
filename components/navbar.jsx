import React from 'react';
import styled from 'styled-components';
import NavbarItem from './navbar-item';

export const Navbar = () => (
  <StyledNavbar>
    <StyledTitle>Giffer</StyledTitle>
    <StyledNavigation>
      <NavbarItem name="Home" link="/" />
      <NavbarItem name="About" link="/about" />
    </StyledNavigation>
  </StyledNavbar>
);

const StyledNavbar = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  flex-wrap: wrap;
  background-color: #4578cc;
  width: 100%;
  z-index: 0;
`;

const StyledTitle = styled.p`
  font-size: 22px;
  justify-self: flex-start;
  margin: 0px;
  margin-left: 5px;
  font-weight: bold;
  color: white;
  user-select: none;
`;

const StyledNavigation = styled.div`
  display: flex;
  flex-wrap: wrap;
`;

export default Navbar;
