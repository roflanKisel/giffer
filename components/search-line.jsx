import React, { PureComponent } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

export class SearchLine extends PureComponent {
  static propTypes = {
    handleSearchByQuery: PropTypes.func.isRequired,
  };

  state = {
    inputText: '',
  };

  handleInputChange = event => {
    this.setState({
      inputText: event.target.value,
    });
  };

  performSearch = event => {
    if (event.key === 'Enter') {
      const { handleSearchByQuery } = this.props;

      handleSearchByQuery({ searchQuery: this.state.inputText });
    }
  };

  render() {
    const { inputText } = this.state;

    return (
      <StyledSearchLine>
        <SearchLineWrapper>
          <h1>Explore</h1>
          <StyledInput
            value={inputText}
            onChange={this.handleInputChange}
            onKeyPress={this.performSearch}
            placeholder="Type something and press enter"
          />
        </SearchLineWrapper>
      </StyledSearchLine>
    );
  }
}

const StyledSearchLine = styled.div`
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  margin-bottom: 20px;
  z-index: 0;
`;

const SearchLineWrapper = styled.div`
  width: 90%;
`;

const StyledInput = styled.input`
  width: calc(100% - 10px);
  height: 50px;
  font-size: 30px;
  padding-left: 10px;
  margin: 0px;
`;

export default SearchLine;
