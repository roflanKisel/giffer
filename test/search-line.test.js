import React from 'react';
import { shallow, mount } from 'enzyme';
import { SearchLine } from '../components/search-line';

describe('SearchLine', () => {
  describe('SearchLine DUMB component', () => {
    it('should renders correctly errors', () => {
      const wrapper = shallow(<SearchLine handleSearchByQuery={() => {}} />)
  
      expect(wrapper).toMatchSnapshot();
    });

    it('should receive all props', () => {
      const wrapper = mount(<SearchLine handleSearchByQuery={() => {}} />);

      expect(wrapper.prop('handleSearchByQuery')).toBeTruthy();
    });
  
    it('should have state', () => {
      const wrapper = mount(<SearchLine handleSearchByQuery={() => {}} />);
  
      expect(wrapper.state().inputText).toEqual('');
    });
  
    it('should change state on inputing', () => {
      const wrapper = mount(<SearchLine handleSearchByQuery={() => {}} />);
  
      expect(wrapper.state().inputText).toEqual('');
      wrapper.find('input').simulate('change', { target: { value: 'a' } });
      expect(wrapper.state().inputText).toEqual('a');
      wrapper.find('input').simulate('change', { target: { value: 'ab' } });
      expect(wrapper.state().inputText).toEqual('ab');
    });
  
    it('should perform action when enter was pressed', () => {
      const wrapper = mount(<SearchLine handleSearchByQuery={() => {}} />)
      const spy = jest.spyOn(wrapper.instance(), 'performSearch');
  
      expect(spy.mock.calls.length).toEqual(0);
      wrapper.find('input').simulate('change', { target: { value: 'a' } });
      expect(spy.mock.calls.length).toEqual(0);
      wrapper.find('input').simulate('keypress', { key: 'Enter' });
      expect(spy.mock.calls.length).toEqual(1);
    });
  });
});