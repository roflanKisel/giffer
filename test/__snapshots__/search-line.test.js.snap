// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchLine SearchLine DUMB component should renders correctly errors 1`] = `
<search-line__StyledSearchLine>
  <search-line__SearchLineWrapper>
    <h1>
      Explore
    </h1>
    <search-line__StyledInput
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Type something and press enter"
      value=""
    />
  </search-line__SearchLineWrapper>
</search-line__StyledSearchLine>
`;
